manifests/install.pp

```diff
@@ -114,4 +114,15 @@
       system => true,
     }
   }
+
+  if ($consul::bash_completion) and ($consul::install_method != 'docker' ) {
+      file { '/etc/bash_completion.d/consul':
```

Comment:
I'm wondering if this directory is correct? shouldn't it go into /etc/profile.d/? That works on Ubuntu/Debian/Arch/RedHat (I haven't tested others).

Reply:
Hi @bastelfreak 

thanks for the input.

The package bash-completion provides `/etc/profile.d/bash_completion.sh` which calls `/usr/share/bash-completion/bash_completion`. The later one then checks `/etc/bash_completion.d/` for any completion scripts.
The drawback I found is that `/etc/bash_completion.d/` is not created by the package bash-completion (at least version 2.8-6 which we run). Other packages do so like git, unp or what so ever.

If the completion should get realized via `/etc/profile.d/` another file content has to be used. 
How did you deploy such completions in your landcape?

Reply:
ah you're right. /etc/profile.d doesn't work here because `complete` is a bash builtin. the code contains stuff like `compat_dir=${BASH_COMPLETION_COMPAT_DIR:-/etc/bash_completion.d}` so I Ithink `/etc/bash_completion.d/` is indeed correct and my initial comment was just stupid. I think `/etc/bash_completion.d` should be configureable in case people have set BASH_COMPLETION_COMPAT_DIR. Those people will want to set the same path within this puppet module.

Reply:
`$bash_completion_compat_dir` has been added together with the ensure block.

If this works on other [os variants](https://github.com/solarkennedy/puppet-consul/blob/master/manifests/params.pp#L63) we may spread `$bash_completion = true`. Unfortunately I have no other distribution available to test this.